test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+var assert = require('assert');
+
+var when = require('when');
+
+describe('bricks-net-api', function () {
+	var api = require('../lib/bricks-net-api');
+	var HTTPRedirectNotAllowedException = require('../lib/bricks-net-exceptions').HTTPRedirectNotAllowedException;
+	var HTTPRedirectLoopException = require('../lib/bricks-net-exceptions').HTTPRedirectLoopException;
+	var ConnectionResetByPeer = require('../lib/bricks-net-exceptions').ConnectionResetByPeer;
+	
+	it('should export `GET` function', function () {
+		assert.equal('function', typeof api.GET);
+	});
+	
+	it('should export `POST` function', function () {
+		assert.equal('function', typeof api.POST);
+	});
+	
+	it('should export `POSTFromFile` function', function () {
+		assert.equal('function', typeof api.POSTFromFile);
+	});
+	
+	it('should export `HTTP` function', function () {
+		assert.equal('function', typeof api.HTTP);
+	});
+	
+	it('should export `HTTPResponse` function', function () {
+		assert.equal('function', typeof api.HTTPResponse);
+	});
+	
+	describe('`GET` function', function () {
+		it('should return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url") instanceof api.GET);
+		});
+		
+		it('should set `url`', function () {
+			assert.equal('url', api.GET("url").url);
+		});
+		
+		it('should `AllowRedirects` return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url").AllowRedirects() instanceof api.GET);
+		});
+		
+		it('should `AllowRedirects` set `allow_redirects`', function () {
+			assert.equal(true, api.GET("url").AllowRedirects().allow_redirects);
+			assert.equal(false, api.GET("url").AllowRedirects(false).allow_redirects);
+			assert.equal(true, api.GET("url").AllowRedirects(true).allow_redirects);
+		});
+		
+		it('should `UserAgent` return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url").UserAgent("USERAGENT") instanceof api.GET);
+		});
+		
+		it('should `UserAgent` set `custom_user_agent`', function () {
+			assert.equal("USERAGENT", api.GET("url").UserAgent("USERAGENT").custom_user_agent);
+			assert.equal("", api.GET("url").UserAgent("").custom_user_agent);
+		});
+		
+		it('should `UserAgent` throw if argument not set', function () {
+			assert.throws(function () {
+				api.GET("url").UserAgent();
+			});
+		});
+	});
```

Comment:
No tests for invalid arguments for `GET`? (Or just `POST` below doesn't have much to test?) :)